tools/tffarmer/farm.go

```diff
@@ -2,42 +2,106 @@ package main
 
 import (
 	"fmt"
+	"strings"
 
 	"github.com/threefoldtech/zos/pkg/schema"
 	"github.com/threefoldtech/zos/tools/explorer/models/generated/directory"
 	"github.com/urfave/cli"
 )
 
-func registerFarm(c *cli.Context) error {
+func registerFarm(c *cli.Context) (err error) {
 	name := c.Args().First()
 	if name == "" {
 		return fmt.Errorf("farm name needs to be specified")
 	}
 
-	addrs := c.StringSlice("address")
-	assets := c.StringSlice("asset")
+	addrs := c.StringSlice("addresses")
 	email := c.String("email")
 	iyo := c.String("iyo_organization")
 
 	addresses := make([]directory.WalletAddress, len(addrs))
 	for i := range addrs {
-		addresses[i].Address = addrs[i]
-		addresses[i].Asset = assets[i]
+		addresses[i].Asset, addresses[i].Address, err = splitAddressCode(addrs[i])
+		if err != nil {
+			return err
+		}
 	}
 
-	farmID, err := db.FarmRegister(directory.Farm{
+	farm := directory.Farm{
 		Name:            name,
 		ThreebotId:      int64(userid.ThreebotID),
 		Email:           schema.Email(email),
 		IyoOrganization: iyo,
 		WalletAddresses: addresses,
-	})
+	}
+
+	farm.ID, err = db.FarmRegister(farm)
 	if err != nil {
 		return err
 	}
 
-	fmt.Println("Farm registered successfully")
-	fmt.Printf("Name: %s\n", name)
-	fmt.Printf("Farm ID: %d\n", farmID)
+	fmt.Println("Farm updated successfully")
```

Comment:
why does this say "updated" in the register handler?

Reply:
because of copy pastaaaaaa